### DISCUSSION TITLE
automatically set debugLabel based on pattern/atom's name

### DISCUSSION THREAD

**Discussion by wootra** [pmndrs/jotai#1981] [Ideas]
Hello, team.
while atom.debugLabel is useful for the debug, it become very tedious to put every atoms.
(in my team, adding debugLabel to every atom is a convention).
There was a lot of unnecessary time consuming to feedback about missing debugLabel.

I'd like to suggest a pattern to setup debugLabel instead of adding debugLabel for every atom.

in my team, we are trying to put all atoms in one separated file, hooks for it is also in the other file.
so it looks like this:
```javascript
// in atoms.js
export const anAtom1 = atom(0);
anAtom1.debugLabel = "group1/anAtom1";

export const anAtom2 = atom(1);
anAtom2.debugLabel = "group1/anAtom2";
```

while the needed code is just adding atom value, as you see above, we needed more typing only for debugLabel.
I find it is very tedious, so I'd like introduce one additional function to setup debugLabel automatically.

```javascript
import type { Atom } from 'jotai';

export const atomGroupWithDebugName = <
    T extends { [name: string]: Atom<unknown> }
>(
    obj: T,
    format = (atomName: string) => atomName
) => {
    Object.keys(obj).forEach(key => {
        const atom = obj[key];
        atom.debugLabel = format(key);
    }, '');
    return obj;
};
```

and using this function, above atom definitions can be like this:
```javascript
// in atoms.js
export const anAtom1 = atom(0);
export const anAtom2 = atom(1);

atomGroupWithDebugName({anAtom1, anAtom2}, name=>`group1/${name}`);
```

or this pattern is also possible
```javascript
// in atoms.js
const anAtom1 = atom(0);
const anAtom2 = atom(1);

export default atomGroupWithDebugName({anAtom1, anAtom2}, name=>`group1/${name}`);
```

what do you think about this approach?

**Comment by dai-shi** [comment]
That is one solution.

> (in my team, adding debugLabel to every atom is a convention).

Another would be something like this?
```js
import { atom as origAtom } from 'jotai'

export atom = (debugLabel, ...args) => {
  const a = origAtom(...args)
  a.debugLabel = debugLabel
  return a
}
```

**Comment by wootra** [comment]
thank you for the response. 
But it is a bit different from the purpose of the function that is introduced. 
the intention is to reduce manual typing of debugLabel for multiple atoms in the file.
I wonder if there is other use case that is not setting debugLabel individually.
